From 714391e55b4297f8da7fa6c58361aa3c30c0e0c5 Mon Sep 17 00:00:00 2001
From: Fedor Isakov <fedor.isakov@jetbrains.com>
Date: Thu, 25 Jun 2026 09:54:01 +0200
Subject: [PATCH 01/14] MRI-4598 mps-platform: ensure 'build.number' system
 property is passed to Bazel

System properties are not inherited by processes spawned neither by Ant, nor Bazel by default.
Property 'build.number' must be passed to Bazel to keep reproducibility.

Also enable verbose failure reports.
---
 community/mps/build/build/build.xml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/community/mps/build/build/build.xml b/community/mps/build/build/build.xml
index 3e68825866528..81237c8e1ed10 100644
--- a/community/mps/build/build/build.xml
+++ b/community/mps/build/build/build.xml
@@ -46,7 +46,10 @@
     <exec failonerror="true" executable="${platform.root}/bazel.cmd" dir="${platform.root}">
       <arg value="run"/>
       <arg value="//mps/build:mps_builder"/>
+      <arg value="--verbose_failures"/>
+      <arg value="--worker_verbose"/>
       <arg value="--"/>
+      <arg value="--jvm_flag=-Dbuild.number=${build.number}"/>
       <arg value="${platform.root}"/>
     </exec>
   </target>
-- 
2.55.0

